### ISSUE TITLE
HW3 Feedback

### ISSUE THREAD

**Issue by lemonsoup** [asneider/GADataScience30#7]
Good work Amanda!  Comments by section below.

**Give an overall summary of your performance from HW2 on the dataset**
Again great use of supporting functions such as SummaryRow, efficient way to take advantage of the way sklearn is set up such that various models can be run using the same process.

**Do you see any areas for improvement?**
* Also take a look at things like feature engineering and dropped vs imputed data as potential ways to improve score.
* Since BP and age have minimal 0 values, keeping them tends to boost the performance of a lot of the models despite the correlation to other features.

**Run k-Means on the dataset and describe your results**
For k-means, since you are not doing any prediction, use scoring methods which are specifically designed to measure how well your data "clustered" -- inertia and silhouette scores from the lab are good ones to try.

**Run PCA on the dataset and describe your results**
* make sure you only do fit_transform() on training data, and follow with transform() on the test data.  This will ensure that your test data has been transformed the same way as your training data, keeping your model consistent (which you did correctly for StandardScalar)
* check out the PCA lab or HW3 solutions for the part about choosing the number of components to get to 80% explained variance.  For the Pima dataset, it tends to need about 5 components to get there, so with fewer than that your models might not have enough information to work with.

**Use the cluster outputs you got from running k-means as a new feature. Rerun your "worst" and "best" model from HW2, including this new feature along with your old features. Describe your results.**
* km didn't really improve anybody's model performance by too much so this is not surprising 
* you could try doing km on the original dataset, not the PCA features.  This might be slightly cleaner and possibly improve the performance.  

**Use the PCA outputs you got from running PCA above as your features. Rerun your "worst" and "best" model from HW2, with the PCA features instead of your old features. Describe your results.**
* the really low recall and f1 scores from PCA is most likely coming from doing fit_transform instead of transform on the test set - if you change that bit and re-run, there should be an instant boost! Otherwise your setup is well organized and easy to follow.  Definitely give it a shot.

**Give your conclusions on the use of k-Means and PCA on the Diabetes dataset**
I encourage you to re-run with: dropping 0s in blood pressure and age (rather than dropping the columns) , and doing transform() instead of fit_transform() for your PCA test data.
